Changes for pubstwo-jira-1671 (Storing short form oricid in database):

orosz-usgs opened the pull request:
Moved data normalization routines from PubsUtils to DataNormalaztion utility class
Added DataNormalization.denoramlizeOrcid, recfactored code to have regex seperate from Pattern
Updated PersonContributor to have in the json, the orcid field displayed the long form with https
Validated PersonContributor orcid field with @Pattern
Updated PersonContributorBusService to normalize orcid on create or update
Fixed IT tests to use short form orcid

Commit:
Changes for pubstwo-jira-1671 (Storing short form oricid in database):
Moved data normalization routines from PubsUtils to DataNormalaztion utility class
Added DataNormalization.denoramlizeOrcid, recfactored code to have regex seperate from Pattern
Updated PersonContributor to have in the json, the orcid field displayed the long form with https
Validated PersonContributor orcid field with @Pattern
Updated PersonContributorBusService to normalize orcid on create or update
Fixed IT tests to use short form orcid

Commit:
Changes for pubstwo-jira-1671 (Storing short form oricid in database):
  Changes for codacy

dsteinich commented on src/test/java/gov/usgs/cida/pubs/busservice/PersonContributorBusServiceIT.java:
```diff
@@ -68,11 +70,32 @@ public void addObjectTest() {
+		boolean hasValidationMess = person.getValidationErrors() != null && !person.getValidationErrors().isEmpty();
+		assertTrue("Expected a validation error message", hasValidationMess);
+
+		boolean isValidIsFalse = !person.isValid();
```
Please use assertFalse than using a double negative.

dsteinich commented on src/test/java/gov/usgs/cida/pubs/busservice/PersonContributorBusServiceIT.java:
```diff
@@ -68,11 +70,32 @@ public void addObjectTest() {
+
+		assertTrue("Expected id not to be set: " + person.getId(), person.getId() == null || person.getId() == 0);
+
+		boolean hasValidationMess = person.getValidationErrors() != null && !person.getValidationErrors().isEmpty();
```
Please check for the actual message you are expecting - or at least the field.

dsteinich commented on src/main/java/gov/usgs/cida/pubs/utility/DataNormalizationUtils.java:
```diff
@@ -0,0 +1,79 @@
+	
+	/** Translate filter values to format expected when querying against the database.
+	 **/
+	public static void normalizeFilters(Map<String, Object> filters) {
```
After more thought, why not just normalize the orcid in the MvcService line 48 setting of it in the filter? Then we aren't modifying the filter map after the fact.

dsteinich commented on src/main/java/gov/usgs/cida/pubs/domain/PersonContributor.java:
```diff
@@ -98,6 +102,13 @@ public String getOrcid() {
+		String formattedOrcid = DataNormalizationUtils.denormalizeOrcid(orcid);
+		return formattedOrcid == null ? orcid : formattedOrcid;
+	}
+
```
Would it work to change the setOrcid to call DataNormalizationUtils.normalizeOrcid, rather than passing the entire PersonContributor to a normalize function later?

dsteinich commented on src/main/java/gov/usgs/cida/pubs/busservice/PersonContributorBusService.java:
```diff
@@ -45,6 +46,7 @@
 				Contributor<PersonContributor<UsgsContributor>> uc = (UsgsContributor) object;
 				uc.setValidationErrors(validator.validate(uc));
 			}
+			DataNormalizationUtils.normalize(object);
```
The normalizing of the orcid should take place before calling validate. Perhaps this is why the regex couldn't be reused?

Commit:
Changes for pubstwo-jira-1671 (Storing short form oricid in database):
  Refactored per review.

Commit:
Changes for pubstwo-jira-1671 (Storing short form oricid in database):
  Codacy change

dsteinich commented on src/main/java/gov/usgs/cida/pubs/domain/PersonContributor.java:
```diff
@@ -24,6 +27,9 @@
 @Component
 @ParentExists
 public class PersonContributor<D> extends Contributor<PersonContributor<D>> implements ILookup {
+	public static final String ORCID_VALIDATION_REGEX = ".*" + DataNormalizationUtils.ORCID_REGEX;
```
Now that just the 19 characters are being stored, can the ORCID_VALIDATION_REGEX and the DataNormalizationUtils.ORCID_REGEX be the same value?

orosz-usgs replied:
Good catch, was thinking about that this weekend. :-)

Commit:
Changes for pubstwo-jira-1671 (Storing short form oricid in database):
   Feedback from review and weekend pondering:
   Removed the .* from the @Pattern regular expression in PersonContributor
